i18n/en/docusaurus-plugin-content-docs/current/reference/layers.mdx

```diff
@@ -0,0 +1,174 @@
+---
+sidebar_position: 1
+pagination_next: reference/slices-segments
+---
+
+# Layers
+
+Layers are the first level of organisational hierarchy in Feature-Sliced Design. Their purpose is to separate code based on how much responsibility it needs and how many other modules in the app it depends on.
+
+:::note
+
+On this page, a _module_ refers to an internal module in the application — a file or directory with an index file. Not to be confused with npm packages.
+
+:::
+
+Every layer carries special semantic meaning to help you determine how much responsibility you should allocate to a module in your code. The names and meanings of layers are standardized across all projects built with Feature-Sliced Design.
+
+There are 7 layers in total, arranged from most responsibility and dependency to least:
+
+<img alt="" src="/img/layers/layers.svg" width="128" height="240" style={{ float: "right", margin: "0 2em" }} />
+
+1. App
+2. Processes
+3. Pages
+4. Widgets
+5. Features
+6. Entities
+7. Shared
+
+You don't have to use every layer in your project — only add them if you think it brings value to your project.
+
+## Import rule on layers
+
+The difference in dependency is enforced with **the import rule on layers**:
+
+> _A module can only import other modules when they are located on layers strictly below._
+
+For example, a module in `~/features/aaa` cannot import code from `~/features/bbb`, but can import code from `~/entities` and `~/shared`.
+
+## Layer definitions
+
+### Shared
+
+Isolated modules, components and abstractions that are detached from the specifics of the project or business.
+Warning: not to be treated like [a utility dump](https://dev.to/sergeysova/why-utils-helpers-is-a-dump-45fo)!
+
+This layer, unlike others, does not consist of slices, and instead consists of segments directly.
+
+**Content examples**:
+
+* UI kit
+* API client
+* Code working with browser APIs
+* Routing primitives (e.g., authentication-protected routes)
```

Comment:
Protected route always depends on some sort of state or bussiness logic from entities (eg. `isAuthenticated` state flag from '@/entities/session'). 


People can be confused: bcs they cant import entities into shared.

As a solustion, i suggest move authentication-protected routes up to entities/session/model/**

Reply:
Okay, I agree that this isn't a good place for them